eip-5564.md

```diff
@@ -118,49 +120,64 @@ contract IERC5564Messenger {
 }
 ```
 
-An example stealth address scheme for elliptic curves is below. Other stealth schemes may need to modify this approach. **{TODO Modify these steps to match the three function signatures above}**
+An example stealth address scheme for elliptic curves is below. Other stealth schemes may need to modify this approach. 
 
-- $G$ is the generator point of the curve.
+- $G$ represents the generator point of the curve.
 
-- Recipient has private keys $p_{view}$ and $p_{spend}$.
+- Recipient has published a stealth meta-address that constists of the public keys $P_{spend}$ and $P_{view}$
 
-- Recipient publishes corresponding public keys $P_{view}$ and $P_{spend}$ in the `IERC5564Registry`.
+- Recipient has has access to the private keys $p_{spend}$, $p_{view}$ that can access $P_{spend}$ and $P_{view}$.
 
 - Sender generates random 32-byte entropy ephemeral private key $p_{ephemeral}$.
 
-- Sender passes the recipient address and $p_{ephemeral}$ to the `IERC5564Generator` contract's `generateStealthAddress` function.
+- Sender passes the stealth mata-address to the `generateStealthAddress` function.
 
-- This function performs the following computations:
+- The `generateStealthAddress` function performs the following computations:
+  - Generate a random ephemeral private key $p_{ephemeral}$.
+  - Derive the ephemeral public key  $P_{ephemeral}$ from $p_{ephemeral}$.
+  - Parse the spending and viewing public keys, P_{spend} and P_{view}, from the stealth meta-address.
   - A shared secret $s$ is computed as $s = p_{ephemeral} \cdot P_{view}$.
   - The secret is hashed $s_{h} = h(s)$.
-  - The view tag $v$ is extracted by taking the most significant 12 bytes $s_{h}[0:12]$,
-  - Multiplying the shared secret with the generator point $S = s \cdot G$.
+  - The view tag $v$ is extracted by taking the most significant byte $s_{h}[0]$,
+  - Multiply the shared secret with the generator point $S = s \cdot G$.
   - The recipient's stealth public key is computed as $P_{stealth} = P_{spend} + S$.
   - The recipient's stealth address $a_{stealth}$ is computed as $\textrm{pubkeyToAddress}(P_{stealth})$.
+  - The function returns the stealth address $a_{stealth}$, the ephemeral public key $P_{ephemeral}$ and the view tag $v$. 
+
+
+An example function to check whether a stealth address belongs to oneself:
+
+* User has access to the viweing private key $p_{view}$ and the public spending key $P_{spend}$.
+
+* User has access to a set of `Announcements` (ephemeral public keys) and applies the `checkStealthAddress` function to each of them. 
```

Comment:
There's more to an announcement log and it's already defined above, so maybe rephrase to this?
```suggestion
* User has access to a set of `Announcement` events and applies the `checkStealthAddress` function to each of them. 
```